src/main/java/Parser.java

```diff
@@ -0,0 +1,437 @@
+    }
+
+    /**
+     * Communicates with TaskList class and passes current Task object
+     * to TaskList class for marking the Task as done.
+     *
+     * @param index of the particular Task in Task list.
+     */
+    public void markDone(int index) {
+        Task taskSubject = this.tasks.markDone(index);
+        System.out.println("     Nice! I've marked this task as done:\n"
+                + "       " + taskSubject);
+    }
+
+    /**
+     * Communicates with TaskList class and passes current Task object
+     * to TaskList class for removal.
+     *
+     * @param index of the particular Task in Task list.
+     */
+    public void delete(int index) {
+        Task taskSubject = this.tasks.remove(index);
+        System.out.println("     Noted. I've removed this task:\n"
+                + "       " + taskSubject
+                + "\n     Now you have " + this.tasks.getLength() + " task(s) in the list.");
+    }
+
+    /**
+     * A command to close the Parser object for it to stop running.
+     * Sets isProgramRunning to false.
+     */
+    public void close() {
+        this.isProgramRunning = false;
+        System.out.println("     Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * Communicates with TaskList class and to get the current stored Task list.
+     */
+    public List<Task> getTasks() {
+        return tasks.getTasks();
+    }
+
+    /**
+     * Communicates with TaskList class and to get the current stored Task list.
+     * Filters the tasks based on the keyword searched
+     *
+     * @param keyword String will be searched within all tasks
+     * If matched, it will be printed.
+     */
+    public void findTask(String keyword) {
+        List<Task> allTasks = tasks.getTasks();
+        int fullSize = tasks.getLength();
+        List<Task> filteredTasks = new ArrayList<>();
+
+        for (int i = 0; i < fullSize; i++) {
+            Task task = allTasks.get(i);
+            String taskString = task.toString();
+            if (taskString.contains(keyword)) {
+                filteredTasks.add(task);
+            }
+        }
+
+        System.out.println("     Here are the matching task(s) in your list:");
+        int partialSize = filteredTasks.size();
+        int index = 1;
+
+        for (int i = 0; i < partialSize; i++) {
+            System.out.println("     " + index + "." + filteredTasks.get(i));
+            index++;
+        }
+    }
+
+    /**
+     * Processes user input for classification and execution.
+     *
+     * @param str is the line of String obtained from user input.
+     *
+     * Reads and makes sense of user input commands which includes:
+     * (bye, list, todo, deadline, event, delete)
+     *
+     * It then executes the commands.
+     *
+     * @return a String representation of Duke output.
+     *
+     * @throws DukeException when command is wrong, unidentifiable or missing.
+     */
+    public String parseUserInput(String str) throws DukeException {
```

Comment:
Maybe you can consider modifying the javadoc for this method or checkUserInput? The purpose of the functions seems very similar to me.

Reply:
Sure 👍 will do asap in the next commit. Thanks for reviewing!